tests/admin_scripts/tests.py

```diff
@@ -1733,6 +1734,46 @@ def test_skip_checks(self, mocked_check, *mocked_objects):
         self.assertIn("Performing system checks...", self.output.getvalue())
         mocked_check.assert_called()
 
+    def test_custom_system_checks(self):
+        @register(Tags.signals)
+        def my_check(app_configs, **kwargs):
+            return [Error("my error")]
+
+        class CustomException(Exception):
+            pass
+
+        class CustomRunserverCommand(RunserverCommand):
+            """Rather than mock run(), raise immediately after system checks run."""
```

Comment:
`run()` calls `inner_run()`, which calls `check()`--which I didn't want to mock--and `run()`, which I did, so I found this solution to most easily meet my needs. Open to alternatives, of course.